Hakawai/Mentions/HKWMentionsCreationStateMachine.m

```diff
@@ -270,6 +270,14 @@ - (void)stringDeleted:(NSString *)deleteString isControlCharacterDeleted:(BOOL)i
 
     __strong __auto_type delegate = self.delegate;
 
+    BOOL isControlCharacterDeleted = NO;
```

Comment:
can you add a comment explaining this

Reply:
When mentions was originally triggered because the whitespace between a control character and a word was deleted, the cursor is next to the control character (like "@|John", where '|' represents the cursor-state). If the user then deletes the control character, the string buffer will not be empty (it will have "John" in it), but mentions has to stop, because control character is deleted. We use the isControlCharacterDeleted flag to decide what to do in this case of control character deletion.